.github/workflows/on_pull_request.yml

```diff
@@ -24,8 +28,19 @@ jobs:
       - uses: nrwl/nx-set-shas@v4
         with:
           main-branch-name: master
-      - name: run build-all
-        run: pnpm build-all --exclude="examples/**" && exit 0
+      - name: run build affected
+        env:
+          ZE_IS_PREVIEW: true
+          ZE_API: ${{ vars.ZE_API }}
+          ZE_API_GATE: ${{ vars.ZE_API_GATEWAY }}
+          ZE_SECRET_TOKEN: ${{ secrets.ZE_SECRET_TOKEN }}
+        run: pnpm dlx nx affected -t build --skipRemoteCache && exit 0
+      - name: Upload built libs and cache
```

Comment:
why not use regular cache? and cache key should take in account job multiplication like matrix.os

Reply:
Fixed